src/aktualizr_info/main.cc

```diff
@@ -77,219 +78,236 @@ int main(int argc, char **argv) {
       readonly = false;
     }
 
+    bool wait_provisioning = false;
+    if (vm.count("wait-until-provisioned") != 0) {
+      wait_provisioning = true;
+    }
+
     std::shared_ptr<INvStorage> storage = INvStorage::newStorage(config.storage, readonly);
     bool cmd_trigger = false;
+    bool provisioned = false;
+    bool dev_id_loaded = false;
+    bool ecu_id_loaded = false;
     std::string device_id;
 
-    if (!storage->loadDeviceId(&device_id)) {
-      std::cout << "Couldn't load device ID" << std::endl;
-    } else {
-      // Early return if only printing device ID.
-      if (vm.count("name-only") != 0u) {
-        std::cout << device_id << std::endl;
-        return EXIT_SUCCESS;
+    while (!provisioned && wait_provisioning) {
```

Comment:
This will break anything that doesn't use the `--wait-until-provisioned` flag since the loop will get skipped.

Reply:
Yes, I realized that too. I'm fixing it right now.